docs/architecture/README.md

```diff
@@ -3,111 +3,184 @@ order: 3
 
-Kestra is a simple Java application that is provided as an executable. You have many deployments options:
-- [Docker](../administrator-guide/deployment/docker)
-- [Kubernetes](../administrator-guide/deployment/kubernetes)
-- [Manual deployment](../administrator-guide/deployment/manual)
+Each of these internal components is provided on multiple implementations depending on your needs and your deployment architecture. You may need to install additional plugins to use some implementations.
 
-At its heart, Kestra has a pluggable system allowing some dependencies to be switched out, for now there is two different production environment architectures available:
+### The Internal Storage
 
-### Medium-sized environment
-![Kestra Architecture](./architecture-sql.svg "Kestra Architecture")
+Kestra uses an **Internal Storage** to handle incoming and outgoing files of varying sizes. This notion is included in the heart of Kestra. Nowadays, storage availability is backed up by many file systems, we rely on these to guarantee the scalability of Kestra. The *Kestra Internal Storage* will store all the files generated by all tasks and transmit them between different tasks if needed.
 
-For medium-sized deployment where high availability is not a strict requirement, you can use a simple database (Postgres or MySQL) as only dependencies. This allows to run Kestra with a minimal stack to maintain. We have for now 2 databases available:
-- Postgres
-- Mysql
+By default, only the local storage is available, it uses a directory inside the host filesystem so it is not scalable and not designed for production usage.
 
-More detail on configurations are available [here](../administrator-guide/configuration/databases/README.md)
+More implementations are available as plugins.
 
-### High Availability and No Single Point of Failure
+You can replace the local storage by one of the following storage implementations:
+- [Storage Minio](https://github.com/kestra-io/storage-minio) for [Minio](https://min.io/), which is compatible with [AWS S3](https://aws.amazon.com/s3/) and all other *S3 Like* storage.
+- [Storage GCS](https://github.com/kestra-io/storage-gcs) for [Google Cloud Storage](https://cloud.google.com/storage).
+- [Storage Azure](https://github.com/kestra-io/storage-azure) for [Azure Blobs Storage](https://azure.microsoft.com/en-us/services/storage/blobs/).
 
-![Kestra Architecture](./architecture.svg "Kestra Architecture")
+### The Queue
 
-In order to support higher throughput, and full horizontal and vertical scaling of the Kestra cluster, we can replace the database with Kafka and Elasticsearch. In that case, all the process can be scaled without any single point of failure.
+The Queue or more precisely queues are used internally for communication between the different Kestra server components. Kestra provides multiple queue types that must be used with their repository counterparts.
```

Comment:
```suggestion
The queues are used internally for communication between the different Kestra server components. Kestra provides multiple queue types that must be used with their repository counterparts.
```

Reply:
Well, the issue here is that we talk about The Queue, The Repository (and The Internal Storage) as internal Kestra components but in fact there is multiple queues and multiple repositories.